build_scripts/build_aws-sdk-cpp.sh

```diff
@@ -16,33 +16,96 @@
+echo "set(CMAKE_C_COMPILER ${CC_COMP})" >> ${TOOLCHAIN_FILE}
+echo "set(CMAKE_CXX_COMPILER ${CXX_COMP})" >> ${TOOLCHAIN_FILE}
 # only when cross compiling
 if [ "${CC_COMP}" != "gcc" ]; then
-    echo "set(CMAKE_FIND_ROOT_PATH ${INSTALL_PREFIX})" >> toolchain.cmake
-    echo "set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)" >> toolchain.cmake
-    echo "set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)" >> toolchain.cmake
-    echo "set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)" >> toolchain.cmake
+    echo "set(CMAKE_FIND_ROOT_PATH ${INSTALL_PREFIX})" >> ${TOOLCHAIN_FILE}
+    echo "set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)" >> ${TOOLCHAIN_FILE}
+    echo "set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)" >> ${TOOLCHAIN_FILE}
+    echo "set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)" >> ${TOOLCHAIN_FILE}
 fi
-echo "set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")" >> toolchain.cmake
-echo "set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")" >> toolchain.cmake
+echo "set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")" >> ${TOOLCHAIN_FILE}
+echo "set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")" >> ${TOOLCHAIN_FILE}
 
+# Build and install static OpenSSL libs to a temporary dir
+pushd ${ROOT_DIR}/third_party/openssl
+
+CFLAGS="$CFLAGS -fPIC -Wa,--noexecstack"
+declare -a OPTS
+OPTS+=(no-shared)
+OPTS+=(no-tests)
+OPTS+=(--libdir=lib)
+OPTS+=(--prefix=${DEPS_PREFIX})
+OPTS+=(--openssldir=${DEPS_OPENSSLDIR})
+_CONFIGURATOR="./Configure"
+case "$CMAKE_TARGET_ARCH" in
+  x86_64)
+    OPTS+=(linux-x86_64)
+    ;;
+  aarch64)
+    OPTS+=(linux-aarch64)
+    ;;
+esac
+CC=${CC_COMP} CXX=${CXX_COMP}" ${CPPFLAGS} ${CFLAGS}" \
+  ${_CONFIGURATOR} ${OPTS[@]} ${LDFLAGS}
+make -j"$(grep ^processor /proc/cpuinfo | wc -l)"
+make install
+
+popd
+# OpenSSL done
+
+# Build and install static libcurl library to a temporary dir
+pushd ${ROOT_DIR}/third_party/curl/build
 cmake -DCMAKE_BUILD_TYPE=Release \
-      -DCMAKE_TOOLCHAIN_FILE=toolchain.cmake \
+      -DOPENSSL_INCLUDE_DIR="${DEPS_PREFIX}/include" \
+      -DOPENSSL_SSL_LIBRARY="${DEPS_PREFIX}/lib/libssl.a" \
+      -DOPENSSL_CRYPTO_LIBRARY="${DEPS_PREFIX}/lib/libssl.a" \
+      -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN_FILE} \
+      -DCMAKE_INSTALL_PREFIX=${DEPS_PREFIX} \
+      -DBUILD_SHARED_LIBS=OFF \
+      -DBUILD_CURL_EXE=OFF \
+      -DBUILD_STATIC_LIBS=ON \
+      ..
+make -j"$(grep ^processor /proc/cpuinfo | wc -l)"
+make install
+popd
+# libcurl done
+
+# Build AWS SDK libs, link statically with libcurl and OpenSSL
+pushd ${ROOT_DIR}/third_party/aws-sdk-cpp/build
+cmake -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_SYSTEM_INCLUDE_PATH="${DEPS_PREFIX}/include" \
+      -DCMAKE_SYSTEM_PREFIX_PATH="${DEPS_PREFIX}" \
+      -DCURL_INCLUDE_DIR="${DEPS_PREFIX}/include" \
+      -DCURL_LIBRARY_RELEASE="${DEPS_PREFIX}/lib/libcurl.a" \
+      -DCURL_LIBRARIES="${DEPS_PREFIX}/lib/libcurl.a" \
+      -Dcrypto_INCLUDE_DIR="${DEPS_PREFIX}/include" \
+      -Dcrypto_SHARED_LIBRARY="${DEPS_PREFIX}/lib/libcrypto.a" \
```

Comment:
Do we still need to provide the _SHARED_LIBRARY variable?

Reply:
This library doesn't really have a switch for static linking, so I am making sure that there's no "better" shared-object library available for it to link with. I had the situation where it could still find an openssl library in the system and it preferred that one.